### PR TITLE
Add endpoint for getting usage between timestamps for a user

### PR DESCRIPTION
**Pull request by jperata**
Relates to #174 

Endpoint: v1/virtualDeviceUsage
Parameters: (inside body)
- userId (userId from the user in firebase)
- startDate (timestamp in string format)
- endDate (timestamp in string format)

Returns:
- userId
- startDate (timestamp in string format)
- endDate (timestamp in string format)
- processedMessages (integer)

### CODE REVIEW

File: src/main/models/VirtualDeviceStat.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import * as mongoose from "mongoose";
 
+mongoose.set('debug', true);
```
Comment (jricaldi) [inline src/main/models/VirtualDeviceStat.ts]: sure?

File: src/main/services/v1/GetVirtualDeviceUsage.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            const userId = req.body.userId;
+
+            if (!userId) {
+                Returns.NotOkay(res, 400, "userId parameter not present in the body").send();
+                return res;
+            }
+
+            const fbDb = new FirebaseDatabase(db);
+            const virtualDevices = await fbDb.getUserVirtualDevices(userId);
+            console.log("virtualDevices", virtualDevices);
```
Comment (jricaldi) [inline src/main/services/v1/GetVirtualDeviceUsage.ts]: ups

File: src/main/services/v1/GetVirtualDeviceUsage.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import * as Admin from "firebase-admin";
```
Comment (jricaldi) [inline src/main/services/v1/GetVirtualDeviceUsage.ts]: there are console.logs here. They are on purpose?

File: src/main/services/v1/GetVirtualDeviceUsage.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+import * as Express from "express";
+import * as Returns from "./Returns";
+import { FirebaseDatabase } from "../../firebase/FirebaseController";
+import * as dotenv from "dotenv";
+import { MongoClient } from 'mongodb';
+
+dotenv.config();
```
Comment (ecruzado) [inline src/main/services/v1/GetVirtualDeviceUsage.ts]: is this necessary?

File: src/main/services/v1/GetVirtualDeviceUsage.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            if (!userId && !startDateString && !endDateString) {
+                Returns.NotOkay(res, 400, "Missing parameter not present in the body, userId, startDate and endDate required.").send();
+                return res;
+            }
+
+            const fbDb = new FirebaseDatabase(db);
+            const virtualDevices = await fbDb.getUserVirtualDevices(userId);
+
+            if (!virtualDevices) {
+                Returns.Okay(res).send(0);
```
Comment (ecruzado) [inline src/main/services/v1/GetVirtualDeviceUsage.ts]: just returning a plain 0, it is not even a json value

File: src/main/services/v1/GetVirtualDeviceUsage.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+            virtualDevicesKeys.forEach(virtualDeviceKey => {
+                const virtualDevice = virtualDevices[virtualDeviceKey];
+                if (virtualDevice) {
+                    tokensToEvaluate.push(virtualDevice.token);
+                }
+            });
+
+            const gteDate = new Date(startDateString);
+            const lteDate = new Date(endDateString);
+
```
Comment (ecruzado) [inline src/main/services/v1/GetVirtualDeviceUsage.ts]: are you using utc dates?

File: src/main/services/v1/GetVirtualDeviceUsage.ts (hunk 4 above)
Comment (jperata) [reply src/main/services/v1/GetVirtualDeviceUsage.ts]: yes, we read process.env.MONGO_URL and process.env.MONGO_DATABASE_NAME here, dotenv is necessary for developmente

File: src/main/services/v1/GetVirtualDeviceUsage.ts (hunk 6 above)
Comment (jperata) [reply src/main/services/v1/GetVirtualDeviceUsage.ts]: I was using this format: 2018-02-18T05:00:00.000Z

**Pull request merged**